python_modules/dagster/dagster_tests/storage_tests/test_input_manager.py

```diff
@@ -379,6 +383,43 @@ def check_input_managers():
     check_input_managers.execute_in_process()
 
 
+def test_input_manager_with_assets():
+    @asset
+    def upstream() -> int:
+        return 1
+
+    @asset(ins={"upstream": AssetIn(input_manager_key="special_io_manager")})
+    def downstream(upstream) -> int:
+        return upstream + 1
+
+    class MyIOManager(IOManager):
+        def load_input(self, context):
+            assert context.upstream_output is not None
```

Comment:
I think it would also be worthwhile to assert that the right asset key is on the context